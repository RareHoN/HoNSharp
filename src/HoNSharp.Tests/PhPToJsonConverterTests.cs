using ApprovalTests;
using ApprovalTests.Reporters;
using HoNSharp.Converters;
using Newtonsoft.Json;

namespace HoNSharp.Tests
{
    [UseReporter(typeof(DiffReporter))]
    public class PhPToJsonConverterTests
    {
        private readonly PhPToJsonConverter _converter;

        public PhPToJsonConverterTests()
        {
            _converter = new PhPToJsonConverter();
        }

        [Fact]
        public void ConvertToJson_GetAccountAllHeroStats_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("get_account_all_hero_stats_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_GetMatchStats_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("get_match_stats_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_GetMatchStats_WithDateTimeOffsetError_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("get_match_stats_datetimeoffset_error_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_GetMatchStats_WithUnexpectedCharacterError_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("get_match_stats_unexpected_character_error_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_MatchHistoryOverviewMidwars_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("match_history_overview_midwars_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_MatchHistoryOverviewNormal_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("match_history_overview_normal_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_ShowStatsCampaign_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("show_stats_campaign_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_ShowStatsCampaignMidwars_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("show_stats_campaign_midwars_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_ShowStatsMastery_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("show_stats_mastery_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_SimpleStats_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("simple_stats_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        private string ReadFile(string responseFileName)
        {
            var responseFilePath = Path.Combine(Directory.GetCurrentDirectory(), "TestData", responseFileName);

            return File.ReadAllText(responseFilePath);
        }

        private string FormatJson(string json)
        {
            var deserialized = JsonConvert.DeserializeObject(json);
            return JsonConvert.SerializeObject(deserialized, Formatting.Indented);
        }
    }
}