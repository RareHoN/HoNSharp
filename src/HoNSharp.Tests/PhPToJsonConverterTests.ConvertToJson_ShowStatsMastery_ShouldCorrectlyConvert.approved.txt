{
  "account_id": 83079,
  "nickname": "RareHoN",
  "name": "KONGOR",
  "rank": "Leader",
  "standing": 3,
  "create_date": "05/08/2009",
  "last_activity": "20/06/2022",
  "selected_upgrades": {
    "0": "av.Flamboyant",
    "1": "c.cat_courier",
    "2": "t.Dumpster_Taunt",
    "3": "cs.legacy",
    "4": "w.8bit_ward",
    "5": "sc.punk_circle_upgrade",
    "6": "te.Punk TP",
    "7": "cc.limesoda",
    "8": "ai.Eye Heart HoN"
  },
  "level": 258,
  "level_exp": 3503475,
  "mastery_info": {
    "0": {
      "heroname": "Hero_Armadon",
      "exp": 16951
    },
    "1": {
      "heroname": "Hero_Behemoth",
      "exp": 22769
    },
    "2": {
      "heroname": "Hero_Chronos",
      "exp": 16850
    },
    "3": {
      "heroname": "Hero_Defiler",
      "exp": 25132
    },
    "4": {
      "heroname": "Hero_Devourer",
      "exp": 28163
    },
    "5": {
      "heroname": "Hero_DwarfMagi",
      "exp": 7317
    },
    "6": {
      "heroname": "Hero_Ebulus",
      "exp": 36100
    },
    "7": {
      "heroname": "Hero_Electrician",
      "exp": 4705
    },
    "8": {
      "heroname": "Hero_Fairy",
      "exp": 22098
    },
    "9": {
      "heroname": "Hero_Frosty",
      "exp": 5298
    },
    "10": {
      "heroname": "Hero_Hammerstorm",
      "exp": 6662
    },
    "11": {
      "heroname": "Hero_Hantumon",
      "exp": 2512
    },
    "12": {
      "heroname": "Hero_Hiro",
      "exp": 1939
    },
    "13": {
      "heroname": "Hero_Hunter",
      "exp": 3454
    },
    "14": {
      "heroname": "Hero_Kraken",
      "exp": 3071
    },
    "15": {
      "heroname": "Hero_Kunas",
      "exp": 36100
    },
    "16": {
      "heroname": "Hero_Krixi",
      "exp": 6841
    },
    "17": {
      "heroname": "Hero_PollywogPriest",
      "exp": 3165
    },
    "18": {
      "heroname": "Hero_Rocky",
      "exp": 5799
    },
    "19": {
      "heroname": "Hero_Soulstealer",
      "exp": 3909
    },
    "20": {
      "heroname": "Hero_Treant",
      "exp": 5132
    },
    "21": {
      "heroname": "Hero_Vanya",
      "exp": 1067
    },
    "22": {
      "heroname": "Hero_Voodoo",
      "exp": 16224
    },
    "23": {
      "heroname": "Hero_WolfMan",
      "exp": 3850
    },
    "24": {
      "heroname": "Hero_Yogi",
      "exp": 15469
    },
    "25": {
      "heroname": "Hero_Zephyr",
      "exp": 1088
    },
    "26": {
      "heroname": "Hero_Mumra",
      "exp": 36100
    },
    "27": {
      "heroname": "Hero_Tempest",
      "exp": 10878
    },
    "28": {
      "heroname": "Hero_Ophelia",
      "exp": 5437
    },
    "29": {
      "heroname": "Hero_Javaras",
      "exp": 5306
    },
    "30": {
      "heroname": "Hero_Legionnaire",
      "exp": 4398
    },
    "31": {
      "heroname": "Hero_Predator",
      "exp": 12736
    },
    "32": {
      "heroname": "Hero_Accursed",
      "exp": 8636
    },
    "33": {
      "heroname": "Hero_Nomad",
      "exp": 2214
    },
    "34": {
      "heroname": "Hero_Scar",
      "exp": 4725
    },
    "35": {
      "heroname": "Hero_Shaman",
      "exp": 36100
    },
    "36": {
      "heroname": "Hero_Scout",
      "exp": 1156
    },
    "37": {
      "heroname": "Hero_Jereziah",
      "exp": 4726
    },
    "38": {
      "heroname": "Hero_Xalynx",
      "exp": 4591
    },
    "39": {
      "heroname": "Hero_PuppetMaster",
      "exp": 2666
    },
    "40": {
      "heroname": "Hero_Arachna",
      "exp": 8162
    },
    "41": {
      "heroname": "Hero_Hellbringer",
      "exp": 36100
    },
    "42": {
      "heroname": "Hero_Pyromancer",
      "exp": 1856
    },
    "43": {
      "heroname": "Hero_Pestilence",
      "exp": 7040
    },
    "44": {
      "heroname": "Hero_Maliken",
      "exp": 22040
    },
    "45": {
      "heroname": "Hero_Andromeda",
      "exp": 2805
    },
    "46": {
      "heroname": "Hero_Valkyrie",
      "exp": 5619
    },
    "47": {
      "heroname": "Hero_BabaYaga",
      "exp": 2616
    },
    "48": {
      "heroname": "Hero_Succubis",
      "exp": 5252
    },
    "49": {
      "heroname": "Hero_Magmar",
      "exp": 3260
    },
    "50": {
      "heroname": "Hero_DiseasedRider",
      "exp": 8766
    },
    "51": {
      "heroname": "Hero_HellDemon",
      "exp": 8033
    },
    "52": {
      "heroname": "Hero_Panda",
      "exp": 2187
    },
    "53": {
      "heroname": "Hero_CorruptedDisciple",
      "exp": 10363
    },
    "54": {
      "heroname": "Hero_Vindicator",
      "exp": 12851
    },
    "55": {
      "heroname": "Hero_SandWraith",
      "exp": 9114
    },
    "56": {
      "heroname": "Hero_Rampage",
      "exp": 3798
    },
    "57": {
      "heroname": "Hero_WitchSlayer",
      "exp": 2150
    },
    "58": {
      "heroname": "Hero_ForsakenArcher",
      "exp": 16824
    },
    "59": {
      "heroname": "Hero_Engineer",
      "exp": 7321
    },
    "60": {
      "heroname": "Hero_Deadwood",
      "exp": 29396
    },
    "61": {
      "heroname": "Hero_Chipper",
      "exp": 2830
    },
    "62": {
      "heroname": "Hero_Bubbles",
      "exp": 1824
    },
    "63": {
      "heroname": "Hero_Fade",
      "exp": 3495
    },
    "64": {
      "heroname": "Hero_Bephelgor",
      "exp": 12035
    },
    "65": {
      "heroname": "Hero_Gauntlet",
      "exp": 33712
    },
    "66": {
      "heroname": "Hero_Tundra",
      "exp": 9966
    },
    "67": {
      "heroname": "Hero_Gladiator",
      "exp": 11565
    },
    "68": {
      "heroname": "Hero_DoctorRepulsor",
      "exp": 16028
    },
    "69": {
      "heroname": "Hero_FlintBeastwood",
      "exp": 32400
    },
    "70": {
      "heroname": "Hero_Bombardier",
      "exp": 1117
    },
    "71": {
      "heroname": "Hero_Moraxus",
      "exp": 2579
    },
    "72": {
      "heroname": "Hero_Hydromancer",
      "exp": 10949
    },
    "73": {
      "heroname": "Hero_Dampeer",
      "exp": 1860
    },
    "74": {
      "heroname": "Hero_Empath",
      "exp": 36100
    },
    "75": {
      "heroname": "Hero_Aluna",
      "exp": 5752
    },
    "76": {
      "heroname": "Hero_Tremble",
      "exp": 520
    },
    "77": {
      "heroname": "Hero_Silhouette",
      "exp": 2510
    },
    "78": {
      "heroname": "Hero_Flux",
      "exp": 7648
    },
    "79": {
      "heroname": "Hero_Martyr",
      "exp": 36100
    },
    "80": {
      "heroname": "Hero_Ra",
      "exp": 4363
    },
    "81": {
      "heroname": "Hero_Parasite",
      "exp": 478
    },
    "82": {
      "heroname": "Hero_EmeraldWarden",
      "exp": 7664
    },
    "83": {
      "heroname": "Hero_Revenant",
      "exp": 16110
    },
    "84": {
      "heroname": "Hero_MonkeyKing",
      "exp": 2620
    },
    "85": {
      "heroname": "Hero_DrunkenMaster",
      "exp": 2891
    },
    "86": {
      "heroname": "Hero_MasterOfArms",
      "exp": 36100
    },
    "87": {
      "heroname": "Hero_Rhapsody",
      "exp": 3385
    },
    "88": {
      "heroname": "Hero_Geomancer",
      "exp": 8816
    },
    "89": {
      "heroname": "Hero_Midas",
      "exp": 4572
    },
    "90": {
      "heroname": "Hero_Cthulhuphant",
      "exp": 6165
    },
    "91": {
      "heroname": "Hero_Monarch",
      "exp": 14425
    },
    "92": {
      "heroname": "Hero_Gemini",
      "exp": 2688
    },
    "93": {
      "heroname": "Hero_Dreadknight",
      "exp": 11887
    },
    "94": {
      "heroname": "Hero_ShadowBlade",
      "exp": 22131
    },
    "95": {
      "heroname": "Hero_Artesia",
      "exp": 3920
    },
    "96": {
      "heroname": "Hero_Taint",
      "exp": 2231
    },
    "97": {
      "heroname": "Hero_Berzerker",
      "exp": 3435
    },
    "98": {
      "heroname": "Hero_FlameDragon",
      "exp": 8461
    },
    "99": {
      "heroname": "Hero_Kenisis",
      "exp": 12699
    },
    "100": {
      "heroname": "Hero_Gunblade",
      "exp": 2693
    },
    "101": {
      "heroname": "Hero_Blitz",
      "exp": 614
    },
    "102": {
      "heroname": "Hero_Artillery",
      "exp": 20138
    },
    "103": {
      "heroname": "Hero_Ellonia",
      "exp": 17709
    },
    "104": {
      "heroname": "Hero_Riftmage",
      "exp": 1142
    },
    "105": {
      "heroname": "Hero_Plant",
      "exp": 3751
    },
    "106": {
      "heroname": "Hero_Ravenor",
      "exp": 7027
    },
    "107": {
      "heroname": "Hero_Prophet",
      "exp": 3149
    },
    "108": {
      "heroname": "Hero_Rally",
      "exp": 4237
    },
    "109": {
      "heroname": "Hero_Oogie",
      "exp": 6337
    },
    "110": {
      "heroname": "Hero_Solstice",
      "exp": 4240
    },
    "111": {
      "heroname": "Hero_Pearl",
      "exp": 12417
    },
    "112": {
      "heroname": "Hero_Grinex",
      "exp": 3258
    },
    "113": {
      "heroname": "Hero_Lodestone",
      "exp": 12577
    },
    "114": {
      "heroname": "Hero_Bushwack",
      "exp": 10208
    },
    "115": {
      "heroname": "Hero_Salomon",
      "exp": 3881
    },
    "116": {
      "heroname": "Hero_Prisoner",
      "exp": 22240
    },
    "117": {
      "heroname": "Hero_SirBenzington",
      "exp": 977
    },
    "118": {
      "heroname": "Hero_Circe",
      "exp": 5173
    },
    "119": {
      "heroname": "Hero_Klanx",
      "exp": 7816
    },
    "120": {
      "heroname": "Hero_Riptide",
      "exp": 4693
    },
    "121": {
      "heroname": "Hero_Moira",
      "exp": 2159
    },
    "122": {
      "heroname": "Hero_Tarot",
      "exp": 36100
    },
    "123": {
      "heroname": "Hero_Kane",
      "exp": 6147
    },
    "124": {
      "heroname": "Hero_Calamity",
      "exp": 1702
    },
    "125": {
      "heroname": "Hero_Deadlift",
      "exp": 24814
    },
    "126": {
      "heroname": "Hero_Parallax",
      "exp": 8903
    },
    "127": {
      "heroname": "Hero_Skrap",
      "exp": 7470
    },
    "128": {
      "heroname": "Hero_Nitro",
      "exp": 11670
    },
    "129": {
      "heroname": "Hero_KingKlout",
      "exp": 1102
    },
    "130": {
      "heroname": "Hero_Shellshock",
      "exp": 31210
    },
    "131": {
      "heroname": "Hero_Ichor",
      "exp": 5133
    },
    "132": {
      "heroname": "Hero_Adrenaline",
      "exp": 10079
    },
    "133": {
      "heroname": "Hero_Apex",
      "exp": 1872
    },
    "134": {
      "heroname": "Hero_Warchief",
      "exp": 6756
    },
    "135": {
      "heroname": "Hero_Sapphire",
      "exp": 17892
    },
    "136": {
      "heroname": "Hero_Goldenveil",
      "exp": 1636
    },
    "137": {
      "heroname": "Hero_Chi",
      "exp": 36100
    },
    "138": {
      "heroname": "Hero_Mimix",
      "exp": 7682
    }
  },
  "mastery_rewards": {
    "0": {
      "level": 1,
      "alreadygot": true,
      "reward": {
        "product_id": 3609,
        "product_name": "Mastery Boost",
        "product_local_content": "/ui/fe2/store/icons/mastery_boost.tga",
        "quantity": 2,
        "points": 0,
        "mmpoints": 0,
        "tickets": 0
      }
    },
    "1": {
      "level": 2,
      "alreadygot": true,
      "reward": {
        "product_id": 3860,
        "product_name": "Iron Mastery Badge",
        "product_local_content": "/ui/fe2/store/icons/mastery_iron_badge.tga",
        "quantity": 1,
        "points": 0,
        "mmpoints": 0,
        "tickets": 0
      }
    },
    "2": {
      "level": 3,
      "alreadygot": true,
      "reward": {
        "product_id": 0,
        "product_name": "",
        "product_local_content": "",
        "quantity": 0,
        "points": 0,
        "mmpoints": 75,
        "tickets": 0
      }
    },
    "3": {
      "level": 5,
      "alreadygot": true,
      "reward": {
        "product_id": 3609,
        "product_name": "Mastery Boost",
        "product_local_content": "/ui/fe2/store/icons/mastery_boost.tga",
        "quantity": 5,
        "points": 0,
        "mmpoints": 0,
        "tickets": 0
      }
    },
    "4": {
      "level": 7,
      "alreadygot": true,
      "reward": {
        "product_id": 0,
        "product_name": "",
        "product_local_content": "",
        "quantity": 0,
        "points": 0,
        "mmpoints": 100,
        "tickets": 0
      }
    },
    "5": {
      "level": 10,
      "alreadygot": true,
      "reward": {
        "product_id": 3861,
        "product_name": "Bronze Mastery Badge",
        "product_local_content": "/ui/fe2/store/icons/mastery_bronze_badge.tga",
        "quantity": 1,
        "points": 0,
        "mmpoints": 0,
        "tickets": 0
      }
    },
    "6": {
      "level": 15,
      "alreadygot": true,
      "reward": {
        "product_id": 0,
        "product_name": "",
        "product_local_content": "",
        "quantity": 0,
        "points": 0,
        "mmpoints": 150,
        "tickets": 0
      }
    },
    "7": {
      "level": 20,
      "alreadygot": true,
      "reward": {
        "product_id": 3609,
        "product_name": "Mastery Boost",
        "product_local_content": "/ui/fe2/store/icons/mastery_boost.tga",
        "quantity": 10,
        "points": 0,
        "mmpoints": 0,
        "tickets": 0
      }
    },
    "8": {
      "level": 25,
      "alreadygot": true,
      "reward": {
        "product_id": 0,
        "product_name": "",
        "product_local_content": "",
        "quantity": 0,
        "points": 0,
        "mmpoints": 175,
        "tickets": 0
      }
    },
    "9": {
      "level": 30,
      "alreadygot": true,
      "reward": {
        "product_id": 0,
        "product_name": "",
        "product_local_content": "",
        "quantity": 0,
        "points": 0,
        "mmpoints": 0,
        "tickets": 50
      }
    },
    "10": {
      "level": 35,
      "alreadygot": true,
      "reward": {
        "product_id": 4225,
        "product_name": "Silver Mastery Badge",
        "product_local_content": "/ui/fe2/store/icons/mastery_silver_badge.tga",
        "quantity": 1,
        "points": 0,
        "mmpoints": 0,
        "tickets": 0
      }
    },
    "11": {
      "level": 40,
      "alreadygot": true,
      "reward": {
        "product_id": 3609,
        "product_name": "Mastery Boost",
        "product_local_content": "/ui/fe2/store/icons/mastery_boost.tga",
        "quantity": 10,
        "points": 0,
        "mmpoints": 0,
        "tickets": 0
      }
    },
    "12": {
      "level": 45,
      "alreadygot": true,
      "reward": {
        "product_id": 0,
        "product_name": "",
        "product_local_content": "",
        "quantity": 0,
        "points": 0,
        "mmpoints": 0,
        "tickets": 100
      }
    },
    "13": {
      "level": 50,
      "alreadygot": true,
      "reward": {
        "product_id": 0,
        "product_name": "",
        "product_local_content": "",
        "quantity": 0,
        "points": 100,
        "mmpoints": 0,
        "tickets": 0
      }
    },
    "14": {
      "level": 55,
      "alreadygot": true,
      "reward": {
        "product_id": 4605,
        "product_name": "Super boost",
        "product_local_content": "/ui/fe2/store/icons/mastery_super_boost.tga",
        "quantity": 1,
        "points": 0,
        "mmpoints": 0,
        "tickets": 0
      }
    },
    "15": {
      "level": 60,
      "alreadygot": true,
      "reward": {
        "product_id": 4634,
        "product_name": "Gold Mastery Badge",
        "product_local_content": "/ui/fe2/store/icons/mastery_gold_badge.tga",
        "quantity": 1,
        "points": 0,
        "mmpoints": 0,
        "tickets": 0
      }
    },
    "16": {
      "level": 65,
      "alreadygot": true,
      "reward": {
        "product_id": 3609,
        "product_name": "Mastery Boost",
        "product_local_content": "/ui/fe2/store/icons/mastery_boost.tga",
        "quantity": 10,
        "points": 0,
        "mmpoints": 0,
        "tickets": 0
      }
    },
    "17": {
      "level": 70,
      "alreadygot": true,
      "reward": {
        "product_id": 0,
        "product_name": "",
        "product_local_content": "",
        "quantity": 0,
        "points": 0,
        "mmpoints": 0,
        "tickets": 100
      }
    },
    "18": {
      "level": 75,
      "alreadygot": true,
      "reward": {
        "product_id": 4605,
        "product_name": "Super boost",
        "product_local_content": "/ui/fe2/store/icons/mastery_super_boost.tga",
        "quantity": 1,
        "points": 0,
        "mmpoints": 0,
        "tickets": 0
      }
    },
    "19": {
      "level": 80,
      "alreadygot": true,
      "reward": {
        "product_id": 4635,
        "product_name": "Epic Mastery Badge",
        "product_local_content": "/ui/fe2/store/icons/mastery_epic_badge.tga",
        "quantity": 1,
        "points": 0,
        "mmpoints": 0,
        "tickets": 0
      }
    },
    "20": {
      "level": 85,
      "alreadygot": true,
      "reward": {
        "product_id": 3609,
        "product_name": "Mastery Boost",
        "product_local_content": "/ui/fe2/store/icons/mastery_boost.tga",
        "quantity": 10,
        "points": 0,
        "mmpoints": 0,
        "tickets": 0
      }
    },
    "21": {
      "level": 90,
      "alreadygot": true,
      "reward": {
        "product_id": 0,
        "product_name": "",
        "product_local_content": "",
        "quantity": 0,
        "points": 0,
        "mmpoints": 200,
        "tickets": 0
      }
    },
    "22": {
      "level": 95,
      "alreadygot": true,
      "reward": {
        "product_id": 4605,
        "product_name": "Super boost",
        "product_local_content": "/ui/fe2/store/icons/mastery_super_boost.tga",
        "quantity": 1,
        "points": 0,
        "mmpoints": 0,
        "tickets": 0
      }
    },
    "23": {
      "level": 100,
      "alreadygot": true,
      "reward": {
        "product_id": 0,
        "product_name": "",
        "product_local_content": "",
        "quantity": 0,
        "points": 100,
        "mmpoints": 0,
        "tickets": 0
      }
    }
  },
  "vested_threshold": 5,
  "0": true
}